upcase_booklist_app/upcase_booklist_app.py

```diff
@@ -68,5 +71,23 @@ def author(author_id):
     return render_template("author.html", author_object=author_object)
 
 
+@app.route("/users")
+def users():
+    """Show all of the users and basic user details"""
+    return render_template("users.html", users=users_info)
+
+
+@app.route("/users/<int:user_id>")
+def user(user_id):
+    """Display user details"""
+
+    user_object = None
```

Comment:
So this is good and works totally correct, but try using `filter` to return this. It's probably one line of code.

Reply:
@paulghaddad This is the updated code using filter, but I'm getting an empty list.

`user_object = list(filter(lambda x: x==id, users_info))`

Any suggestions here?

This is what's currently in the user view function:

```
user_object = None
    for user in users_info:
        if user['id'] == id:
            user_object = user

    return render_template("user.html", user_object=user_object)
```

Reply:
Wouldn't it be:

```
user_object = list(filter(lambda x: x['id']==id, users_info))
```

since you have to index into the `id` key of each user dict?